# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: caps/OriginAttributes.h
# Commit: 7c4b003d1e48
# Full Hash: 7c4b003d1e48ac2e847191fe36ce4c39b9280d54
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2024-12-18 04:27:40
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 4: Implement the third-party cookie blocking excpetion in CookieService. r=cookie-reviewers,baku,valentin
#   
#   The patch integrates the third-party cookie blocking excpetion in
#   CookieService.
#   
# ==============================================================================

diff -r 36163490219b -r 7c4b003d1e48 caps/OriginAttributes.h
--- a/caps/OriginAttributes.h	Tue Dec 17 21:44:37 2024 +0000
+++ b/caps/OriginAttributes.h	Tue Dec 17 21:44:37 2024 +0000
@@ -141,6 +141,11 @@
                                 nsAString& outScheme, nsAString& outBaseDomain,
                                 int32_t& outPort,
                                 bool& outForeignByAncestorContext);
+
+  // Parse a partitionKey and extract the site from it. Returns false if the
+  // partitionKey cannot be parsed because the format is invalid.
+  static bool ExtractSiteFromPartitionKey(const nsAString& aPartitionKey,
+                                          nsAString& aOutSite);
 };
 
 class OriginAttributesPattern : public dom::OriginAttributesPatternDictionary {